test_rclcpp/test/test_sigterm.py.in

```diff
@@ -0,0 +1,78 @@
+# generated code does not contain a copyright notice
+
+from launch import LaunchDescription
+from launch.actions import ExecuteProcess
+from launch.events.process import SignalProcess
+
+import launch_testing
+import launch_testing.actions
+import launch_testing.asserts
+import launch_testing_ros
+
+import os
+import signal
+import sys
+import unittest
+
+
+def generate_test_description():
+    env = os.environ.copy()
+    env['RCUTILS_CONSOLE_OUTPUT_FORMAT'] = '[{severity}] [{name}]: {message}'
+    launch_description = LaunchDescription()
+    proc_under_test = ExecuteProcess(
+        cmd=['@TEST_EXECUTABLE@'],
+        name='@TEST_EXECUTABLE_NAME@',
+        sigterm_timeout='15',
+        output='screen',
+        env=env,
+    )
+    launch_description.add_action(proc_under_test)
+    launch_description.add_action(launch_testing.util.KeepAliveProc())
+    launch_description.add_action(
+        launch_testing.actions.ReadyToTest()
+    )
+    print('generated launch description!!', file=sys.stderr)
+    return launch_description, locals()
+
+
+class TestExecutableOutput(unittest.TestCase):
+
+    def @TEST_NAME@(self, launch_service, proc_output, proc_info, proc_under_test):
+        output_filter = launch_testing_ros.tools.basic_output_filter(
+            filtered_rmw_implementation='@rmw_implementation@'
+        )
+        proc_output.assertWaitFor(
+            expected_output=launch_testing.tools.expected_output_from_file(
+                path='@TEST_EXECUTABLE_TRIGGER_SHUTDOWN_OUTPUT@'
+            ), process=proc_under_test, output_filter=output_filter, timeout=10
+        )
+        launch_service.emit_event(SignalProcess(
+            signal_number=signal.SIGTERM,
+            process_matcher=lambda x: x == proc_under_test))
+        output_filter = launch_testing_ros.tools.basic_output_filter(
+            filtered_rmw_implementation='@rmw_implementation@'
+        )
+        # proc_output.assertWaitFor(
+        #     expected_output=launch_testing.tools.expected_output_from_file(
+        #         path='@TEST_EXECUTABLE_EXPECTED_OUTPUT@'
+        #     ), process=proc_under_test, output_filter=output_filter
+        # )
+        proc_info.assertWaitForShutdown(proc_under_test, timeout=20)
+
+
+@launch_testing.post_shutdown_test()
+class TestExecutableOutputAfterShutdown(unittest.TestCase):
+
+    def @TEST_NAME@(self, proc_output, proc_info, proc_under_test):
+        output_filter = launch_testing_ros.tools.basic_output_filter(
+            filtered_rmw_implementation='@rmw_implementation@'
+        )
+        launch_testing.asserts.assertInStderr(
+            proc_output,
+            expected_output=launch_testing.tools.expected_output_from_file(
+                path='@TEST_EXECUTABLE_EXPECTED_OUTPUT@'
+            ), process=proc_under_test, output_filter=output_filter
+        )
+        # launch_testing.asserts.assertExitCodes(
+        #     proc_info, launch_testing.asserts.EXIT_SIGTERM, process=proc_under_test)
```

Comment:
@ivanpauno same nit as above.

Reply:
Solved in https://github.com/ros2/system_tests/pull/485/commits/e259da3875ed7e1184a5f82315b95b627dd8c81d.